pkg/controller/infinispan/resources/service/service_controller.go

```diff
@@ -205,24 +212,32 @@ func setupServiceForEncryption(ispn *ispnv1.Infinispan, service *corev1.Service)
 	}
 }
 
-func computeService(ispn *ispnv1.Infinispan) *corev1.Service {
+func computeAdminService(ispn *ispnv1.Infinispan) *corev1.Service {
+	return computeService(ispn.GetAdminServiceName(), consts.InfinispanAdminPortName, "infinispan-service-admin", consts.InfinispanAdminPort, ispn)
```

Comment:
We can expose admin externally with the same address as main Infinispan port, i.e. 11222
For Metrics users port will be the same as for common users

Reply:
We want a different service to the user service, as we never want to expose the admin service externally. It's exclusively for internal k8s usage, i.e. the operator and prometheus monitors.

Reply:
Exposition doesn't make sense for ocp/k8s internal users
Also have a common posts for admin and users could be nice, as service is mapping between Pod internal IP's and exposition

Reply:
I don't fully understand what you're proposing here, can you elaborate?

Reply:
Sorry, was not correctly explain probably. We can introduce two parameters for compute service, i.e. port (this is external service port) and targetPort (this is internal Pod port) which can be mapped for different services 
admin service: 11222 -> 11223
customer service: 11222 -> 11222
Is this is clear?

Reply:
So two distinct services for admin and user, each of which is exposed on port 11222, but the admin service uses `targetPort: 11223`. I think that makes sense, one less port number for the user to worry about :+1:

Reply:
I'm going to remove this after all. It makes the operator internals more complicated as depending on the operation we're trying to perform we need to be careful about which pod we target. Having two distinct ports for admin and user makes things more explicit. The only downside is that the ServiceMonitor has to use port 11223, however I expect most users will copy and paste the `ServiceMonitor` example from our docs anyway.